zipline/pipeline/slice.py

```diff
@@ -0,0 +1,55 @@
+
+from numpy import searchsorted
+
+
+class Slice(object):
+    """
+    Mixin for extracting a single column of a Term's output.
+
+    Parameters
+    ----------
+    term : zipline.pipeline.term.Term
+        The term from which to extract a column of data.
+    asset : zipline.assets.Asset
+        The asset corresponding to the column of `term` to be extracted.
+
+    Notes
+    -----
+    Users should rarely construct instances of `Slice` directly. Instead, they
+    should construct instances via indexing, e.g. `MyFactor()[Asset(24)]`.
+    """
+
+    def __new__(cls, term, asset):
+        return super(Slice, cls).__new__(
+            cls,
+            asset=asset,
+            inputs=[term],
+            window_length=0,
+            mask=term.mask,
```

Comment:
I wonder if we should do `term.mask[asset]` here to keep the mask dims in sync with the term dims?